airflow/providers/mongo/hooks/mongo.py

```diff
@@ -90,15 +85,25 @@ def get_conn(self) -> MongoClient:
             options.update({"ssl_cert_reqs": CERT_NONE})
 
         self.client = MongoClient(self.uri, **options)
-
         return self.client
 
-    def close_conn(self) -> None:
-        """Closes connection"""
-        client = self.client
-        if client is not None:
-            client.close()
-            self.client = None
+    def _create_uri(self) -> str:
+        """
+        Create URI string from the given credentials.
+        :return: URI string.
+        """
+        srv = self.extras.pop("srv", False)
+        scheme = "mongodb+srv" if srv else "mongodb"
+        login = self.connection.login
+        password = self.connection.password
+        if login is not None and password is not None:
+            login = login.encode()
+            password = password.encode()
+            creds = f"{quote_plus(login)}:{quote_plus(password)}@"
+        else:
+            creds = ""
+        port = f":{self.connection.port}" if self.connection.port else ""
+        return f"{scheme}://{creds}{self.connection.host}{port}/{self.connection.schema}"
```

Comment:
Would it be a good idea to use `urlunsplit` instead of string-joining?

Reply:
Agree with you!

Reply:
Implemented `urlunsplit` for creating uri